ci/docker/amazon-linux-2017.03

```diff
@@ -1,10 +1,18 @@
-FROM amazonlinux:2017.03
+FROM public.ecr.aws/amazonlinux/amazonlinux:2018.03
```

Comment:
intentionally leaving the filename out of date with the base image right now to make my life a bit easier in the short term for managing the creation of the replacement CI workflows

why did the tag change? - because I have automation in #159 to re-drives the docker builds from AWS CodeBuild, that DockerHub likes to throttle - however I could not find a :2017.03 tag in the public.ecr.aws gallery - so, next best thing is to use an actually updated tip tag for the original AmazonLinux